src/server/frontend/userstate.js

```diff
@@ -0,0 +1,57 @@
+import config from '../config';
+import Promise from 'bluebird';
+import Immutable from 'immutable';
+
+// Example how initialState, which is the same for all users, is enriched with
+// user state. With state-less Flux, we don't need instances.
+export default function userState() {
+
+  return (req, res, next) => {
+    loadUserData(req).then(loadedData => {
+      req.userState = Immutable.Map().mergeWith(...loadedData);
+      next();
+    });
+  };
+
+}
+
+// Gracefully settle all promises, ignore failed
+function loadUserData(req) {
+  const dataSources = [
+    acceptLanguages(req),
+    loadTodos()
+  ];
+
+  return Promise.settle(dataSources).then(receivedData =>
+    receivedData
+      .filter(promise => promise.isFulfilled())
+      .map(promise => promise.value())
+  );
+}
+
+function acceptLanguages(req) {
```

Comment:
Maybe we could extract these functions into other files and leave only the middleware + `loadUserData` function in here.